tests/flow/test_ts_delete.py

```diff
@@ -120,3 +120,99 @@ def test_bad_del(self):
         self.assertTrue(r.execute_command("SET", "BAD_X", "NOT_TS"))
         with pytest.raises(redis.ResponseError) as excinfo:
             r.execute_command("TS.DEL", "BAD_X", 100, 200)
+
+def test_del_retention_with_rules(self):
+    sample_len = 1010
+    with Env().getClusterConnectionIfNeeded() as r:
+        r.execute_command("ts.create", 'test_key_2', 'RETENTION', 1, 'uncompressed')
+        r.execute_command("ts.create", 'test_key_3', 'uncompressed')
+        r.execute_command('ts.createrule', 'test_key_2', 'test_key_3', 'AGGREGATION', 'avg', 10)
+
+        for i in range(sample_len):
+            assert i == r.execute_command("ts.add", 'test_key_2', i, 1)
+
+        with pytest.raises(redis.ResponseError) as excinfo:
+            r.execute_command("ts.del", "test_key_2", 1, 10)
+
+        r.execute_command("ts.create", 'test_key_4', 'RETENTION', 25, 'uncompressed')
+        r.execute_command("ts.create", 'test_key_5', 'uncompressed')
+        r.execute_command('ts.createrule', 'test_key_4', 'test_key_5', 'AGGREGATION', 'avg', 10)
+
+        for i in range(30):
+            assert i == r.execute_command("ts.add", 'test_key_4', i, 1)
+
+        with pytest.raises(redis.ResponseError) as excinfo:
+            r.execute_command("ts.del", "test_key_4", 9, 10)
+
+
+def test_del_with_rules(self):
```

Comment:
lets have a test case with lots of samples that the deletion range will span to more than 2 buckets.

Reply:
@danni-m I already added deletion of more than 2 buckets, I don't see how more test like this will change anything